vortex-datafusion/src/persistent/format.rs

```diff
@@ -73,12 +75,29 @@ impl FileFormat for VortexFormat {
         objects: &[ObjectMeta],
     ) -> DFResult<SchemaRef> {
         let mut file_schemas = Vec::default();
-        for o in objects {
-            let os_read_at = ObjectStoreReadAt::new(store.clone(), o.location.clone());
-            let initial_read = read_initial_bytes(&os_read_at, o.size as u64).await?;
-            let lazy_dtype = initial_read.lazy_dtype();
-            let s = infer_schema(lazy_dtype.value()?)?;
-            file_schemas.push(s);
+        let mut tasks: JoinSet<DFResult<Schema>> = objects
```

Comment:
https://stackoverflow.com/questions/63589668/how-to-tokiojoin-multiple-tasks In modern Tokio if you’re using it joinset seems like the right thing.

Reply:
Join_all is fine for IO tasks, JoinSet is useful for CPU tasks since you spawn and make progress on many of them at once across different worker threads.

But in this case we're just reading files. join_all is much simpler syntax and less overhead

Reply:
If you spawn the io on other worker like our iodispatcher then you can join all, otherwise use a joinset

Reply:
Note that if we spawned all this work on the IoDispatcher, then the join_all would just be polling a channel for a result, so even less reason to use a JoinSet.

Reply:
`join_all` and other non-tokio related utilities don't participate in Tokio's cooperative scheduling which I expect to push tail latencies up, which is what your profiling showed?